direct/io-bulkfs-parquet/src/main/java/cz/o2/proxima/direct/bulk/fs/parquet/ParquetFileFormat.java

```diff
@@ -0,0 +1,189 @@
+
+  /** Operation column name in parquet file. See {@link Operation} for details. */
+  static final String PARQUET_COLUMN_NAME_OPERATION = "operation";
+
+  /** Compression for parquet file */
+  CompressionCodecName parquetCompressionCodec;
+
+  @Nullable private transient MessageType parquetSchema;
+  private AttributeFamilyDescriptor familyDescriptor;
+  private String attributeNamesPrefix;
+
+  @Override
+  public void setup(AttributeFamilyDescriptor family) {
+    familyDescriptor = family;
+    parquetCompressionCodec =
+        CompressionCodecName.fromConf(
+            Optional.ofNullable(family.getCfg().get(PARQUET_CONFIG_COMPRESSION_KEY_NAME))
+                .map(Object::toString)
+                .orElse(
+                    Optional.ofNullable(family.getCfg().get("gzip"))
+                        .filter(bool -> Boolean.parseBoolean(bool.toString()))
+                        .map(bool -> CompressionCodecName.GZIP.name())
+                        .orElse(null)));
+
+    attributeNamesPrefix =
+        Optional.ofNullable(family.getCfg().get(PARQUET_CONFIG_VALUES_PREFIX_KEY_NAME))
+            .map(Object::toString)
+            .orElse(PARQUET_DEFAULT_VALUES_NAME_PREFIX);
+    log.info("Parquet schema for family {} is {}", family.getName(), getParquetSchema());
+  }
+
+  @Override
+  public Reader openReader(Path path, EntityDescriptor entity) throws IOException {
+    return new ProximaParquetReader(path, entity);
+  }
+
+  @Override
+  public Writer openWriter(Path path, EntityDescriptor entity) throws IOException {
+    log.debug(
+        "Opening parquet writer for entity [{}] with path [{}] and schema: {}",
+        entity.getName(),
+        path,
+        getParquetSchema());
+    return new ProximaParquetWriter(
+        path,
+        getParquetSchema(),
+        attributeNamesPrefix,
+        parquetCompressionCodec,
+        createWriterConfiguration());
+  }
+
+  Configuration createWriterConfiguration() {
+    Configuration conf = new Configuration();
+    Map<String, Object> familyConf = new HashMap<>(familyDescriptor.getCfg());
+    familyConf.putIfAbsent(PARQUET_CONFIG_PAGE_SIZE_KEY_NAME, PARQUET_DEFAULT_PAGE_SIZE);
+    familyConf.putIfAbsent(PARQUET_CONFIG_COMPRESSION_KEY_NAME, parquetCompressionCodec.name());
+
+    familyConf.putIfAbsent(ParquetOutputFormat.BLOCK_SIZE, PARQUET_DEFAULT_PAGE_SIZE);
+    familyConf.forEach(
+        (k, v) -> {
+          if (k.startsWith("parquet.")) {
+            conf.set(k, v.toString());
+          }
+        });
+    return conf;
+  }
+
+  private MessageType getParquetSchema() {
+    if (parquetSchema == null) {
+      parquetSchema = ParquetUtils.createParquetSchema(familyDescriptor);
+    }
+    return parquetSchema;
+  }
+
+  @Override
+  public String fileSuffix() {
+    return "parquet" + parquetCompressionCodec.getExtension();
+  }
+
+  /** Enum used for mapping proxima operation into (and from) parquet files. */
+  enum Operation {
```

Comment:
nit: enums should be defined under `static` variables section

Reply:
nested enum types are implicitly static

Reply:
yes, I just wanted to move this block up in the file, where the static definition belongs :)